File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.py
import sys

input = sys.stdin.readline

n, m = map(int, input().split())
temp = list((map(int, input().split())))
list = [0, *temp]

for i in range(n + 1):
    list[i] = list[i - 1] + list[i]

for i in range(m):
    a, b = map(int, input().split())
    print(list[b] - list[a - 1])

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		String[] input = br.readLine().split(" ");

		int n = Integer.parseInt(input[0]);
		int m = Integer.parseInt(input[1]);

		input = br.readLine().split(" ");
		int[] array = new int[n];

		for (int i = 0; i < n; i++) {
			array[i] = Integer.parseInt(input[i]);
		}
		int[] sum = new int[n + 2];

		sum[1] = array[0];

		for (int i = 2; i <= n; i++) {
			sum[i] = sum[i - 1] + array[i - 1];
		}

		for (int k = 0; k < m; k++) {
			input = br.readLine().split(" ");
			int i = Integer.parseInt(input[0]);
			int j = Integer.parseInt(input[1]);

			System.out.println(sum[j] - sum[i - 1]);
		}

	}
}